include/arch/common/addr_types.h

```diff
@@ -10,8 +10,8 @@
 #define ZEPHYR_INCLUDE_ARCH_X86_ADDR_TYPES_H_
 
 #ifndef _ASMLANGUAGE
-typedef unsigned int paddr_t;
-typedef unsigned int vaddr_t;
+typedef uintptr_t paddr_t;
+typedef void *vaddr_t;
```

Comment:
why they are different ? can't you define both `uintptr_t` ?

Reply:
I asked her to do it this way. I think it is better to store virtual addresses as `void *` since they represent actual addressable memory locations, and `uintptr_t` for physical memory addresses since they should not be directly dereferenced without a conversion to physical address first. I think there is value in not having the same data types for both as it will be less easy to accidentally confuse the two.

Reply:
> `uintptr_t` for physical memory addresses since they should not be directly dereferenced without a conversion to physical address first.

Did you mean, without a conversion to virtual address first?

> The sys_cache_xxx names now don't reflect i or d. Add a 'd'?

I think we can use `sys_dcache_flush`, `sys_dcache_clean` in this PR to reflect that these are dcache operations(as already suggested by @ruuddw above). Also `sys_dcache_clean` should be considered to be added- if not here, in separate PR. icache APIs can be considered separately anyway.

Reply:
> > `uintptr_t` for physical memory addresses since they should not be directly dereferenced without a conversion to physical address first.
> 
> Did you mean, without a conversion to virtual address first?

Oops, yes that is what I meant.